parse.go

```diff
@@ -104,17 +104,12 @@ type ParseOptions struct {
 	// If true, checks that directives have a valid number of arguments.
 	SkipDirectiveArgsCheck bool
 
-	// MatchFuncs are called in order when an unknown or non-core NGINX directive is
-	// encountered by the parser to determine the valid contexts and argument count of the
-	// directive. Set this option to enable parsing of directives belonging to non-core or
-	// dynamic NGINX modules that follow the usual grammar rules of an NGINX configuration.
-	// If DirectiveSources was provided, this will be ignored.
-	MatchFuncs []MatchFunc
-
 	// An array of MatchFunc, which is used to indicate the OSS/N+ version, and
 	// dynamic modules you want to include for validation. If a directive matches
 	// any of them, and satisfies the corresponding bitmask, it should pass the validation.
-	// If we provide this, MatchFuncs will be ignored
+	// If this is provided, it will be the only source of directives included for validation.
+	// If this is not provided, `defaultDirectives`(NPlus, Njs, and Otel in it) will be included
+	// for validation.
```

Comment:
Generally doc comments should generally have the exported name in the first line. I think this is a generally useful convention as it results in the first line also appear when searching for an exported member providing a little context:

```
$ grep "MatchFunc" parse.go
// MatchFunc is the signature of the match function used to identify NGINX directives that
type MatchFunc func(directive string) (masks []uint, matched bool)
	// MatchFuncs are called in order when an unknown or non-core NGINX directive is
	MatchFuncs []MatchFunc
```

Something like:

```go
	// DirectiveSources is a list of MatchFuncs used to validate directives 
	// encountered while parsing an NGINX configuration... 
```

The current comment mentions `defaultDirectives`-- probably you want to omit the backticks from the comment-- I think you want to mention the exported `DefaultDirectivesMatchFunc` function instead as the unexported variable is not useable from outside the package.